File: 9_palindrome_number.c
/**
 * Description : Palindrome Number
 *               Determine whether an integer is a palindrome. Do this without
 *               extra space.
 * Author      : Evan lau
 * Date        : 2015.10.23
 */

#include <stdio.h>
#include <stdbool.h>

bool isPalindrome(int x)
{
    int y = 0;
    int z = x;
    
    if (x < 0)
        return false;

    while (z != 0)
    {
        y = y * 10 + z % 10;
        z = z / 10;
    }

    if (y == x)
        return true;
    else
        return false;
}

int main()
{
    int n = -1321;

    if (isPalindrome(n))
        printf("Yes\n");
    else
        printf("No\n");
    
    return 0;
}

File: typescript/9_palindrome_number.ts
function isPalindrome(x: number): boolean {
  if (x < 0) return false;
  if (x < 10) return true;

  const s = x.toString();
  let head = 0;
  let tail = s.length - 1;
  while (head < tail) {
    if (s[head++] !== s[tail--]) {
      return false;
    }
  }
  return true;
}
